### PR TITLE
Remove optional from core docs for 'Type'

### LINKED ISSUE #87683: API Docs for corev1.LimitRangeItem show type as +optional and omitempty
**What happened**:
updating controller and using API docs shows corev1.LimitRangeItem as not requiring `Type`, this is not the case.

**What you expected to happen**:
API Docs should show that Type is required

**How to reproduce it (as minimally and precisely as possible)**:
Per API Docs https://github.com/kubernetes/kubernetes/blob/ae95a4bfcac12bf6b19e2d5acf6404359b1a8c3b/staging/src/k8s.io/api/core/v1/types.go#L5211 Type is optional and omitempty
this should work...
```
	return &corev1.LimitRange{
		ObjectMeta: metadata,
		Spec: corev1.LimitRangeSpec{
			Limits: []corev1.LimitRangeItem{
				{
					Max: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("4")},
					Default: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2.5")},
					DefaultRequest: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("1")},
				},
			},
		},
	}
```

This is the required format with Type:
```
	return &corev1.LimitRange{
		ObjectMeta: metadata,
		Spec: corev1.LimitRangeSpec{
			Limits: []corev1.LimitRangeItem{
				{
                                        Type: "Container",
					Max: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2.5")},
					Default: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2.5")},
					DefaultRequest: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2.5")},
				},
			},
		},
	}
```

**Anything else we need to know?**:
Nope.

**Environment**:
- Kubernetes version (use `kubectl version`): 1.14.9
- Cloud provider or hardware configuration: aws EKS
- Others:

### PR DESCRIPTION
**Pull request by bismitaguha**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Modifies the `core/api` docs for the `Type` variable which is not optional.

**Which issue(s) this PR fixes**:

Fixes #87683 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

This is the default format:
```release-note
return &corev1.LimitRange{
		ObjectMeta: metadata,
		Spec: corev1.LimitRangeSpec{
			Limits: []corev1.LimitRangeItem{
				{
                                        Type: "Container",
					Max: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2.5")},
					Default: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2.5")},
					DefaultRequest: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2.5")},
				},
			},
		},
	}
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Comment by bismitaguha** [comment]
CLA done! @k8s-ci-robot

**Comment by bismitaguha** [comment]
I signed it

**Comment by bismitaguha** [comment]
I signed it

**Comment by munnerz** [comment]
I added a comment to the original issue you opened regarding this PR 😄 https://github.com/kubernetes/kubernetes/issues/87683#issuecomment-585706685

**Comment by liggitt** [comment]
Thanks for the PR. This looks good to me (see https://github.com/kubernetes/kubernetes/issues/87683#issuecomment-585759392 for context).

You will need to regenerate the protobuf and openapi docs for this API using `make update`, then commit the results.

**Comment by bismitaguha** [comment]
@liggitt I have run `make update` and `make verify` and committed the results. Please review the changes.

File: staging/src/k8s.io/api/apps/v1/types_swagger_doc_generated.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func (DaemonSetSpec) SwaggerDoc() map[string]string {
 }
 
 var map_DaemonSetStatus = map[string]string{
-	"":                       "DaemonSetStatus represents the current status of a daemon set.",
+	"": "DaemonSetStatus represents the current status of a daemon set.",
```
Comment (liggitt) [inline staging/src/k8s.io/api/apps/v1/types_swagger_doc_generated.go]: all of the whitespace changes are unexpected... what go version are you running with? try running `hack/update-gofmt.sh` before committing

Comment (bismitaguha) [reply staging/src/k8s.io/api/apps/v1/types_swagger_doc_generated.go]: go1.13.7

Comment (bismitaguha) [reply staging/src/k8s.io/api/apps/v1/types_swagger_doc_generated.go]: @liggitt I have properly run all the above mentioned scripts and committed the changes. Please review.

**Comment by liggitt** [comment]
running hack/update-openapi-spec.sh I see this diff:
```diff
diff --git a/api/openapi-spec/swagger.json b/api/openapi-spec/swagger.json
index a65cdf28987..54500e1d50e 100644
--- a/api/openapi-spec/swagger.json
+++ b/api/openapi-spec/swagger.json
@@ -6955,6 +6955,9 @@
           "type": "string"
         }
       },
+      "required": [
+        "type"
+      ],
       "type": "object"
     },
     "io.k8s.api.core.v1.LimitRangeList": {
```

**Comment by bismitaguha** [comment]
@liggitt I guess I have been able to resolve the problem occurring with my PR previously. Please review.

**Pull request merged**